sw/device/silicon_creator/rom/rom_start.S

```diff
@@ -202,7 +197,89 @@ _rom_start_boot:
   li   a0, TOP_EARLGREY_CLKMGR_AON_BASE_ADDR
   sw   t0, CLKMGR_JITTER_ENABLE_REG_OFFSET(a0)
 
-.L_ast_init_skip:
+.L_ast_init_end:
+  // Check if we are trying to transition to RMA. If so, busy loop for
+  // CREATOR_SW_CFG_RMA_SPIN_CYCLES to let the transition start and continue
+  // looping while lc_ctrl is not ready. Reset if the CPU is still executing at
+  // the end.
+  li   a0, (TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR + \
+            OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET)
+  lw   t0, OTP_CTRL_PARAM_CREATOR_SW_CFG_RMA_SPIN_EN_OFFSET(a0)
+  li   t6, HARDENED_BOOL_TRUE
+  bne  t0, t6, .L_rma_spin_skip
+
+  li   a1, (TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR + \
+            OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET)
+  lw   t1, OTP_CTRL_PARAM_CREATOR_SW_CFG_RMA_SPIN_EN_OFFSET(a1)
+  beq  t1, t6, .L_rma_spin_init
+  unimp
+  unimp
+  unimp
+  unimp
+
+.L_rma_spin_init:
+  csrw mcycle, zero
+  li   a0, TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR
+  sw   zero, AON_TIMER_WDOG_CTRL_REG_OFFSET(a0)
+
+  // Configure strap pins.
+  //
+  // | GPIO Pin:      | 24        | 23        | 22      |
+  // |----------------+-----------+-----------+---------|
+  // | Configuration: | Input     | Input     | Input   |
+  // |                | Pull-down | Pull-down | Pull-up |
```

Comment:
Looking at the code below, I think it's pin 24 that has pull-up rather than pull-down:
```suggestion
  // | GPIO Pin:      | 22        | 23        | 24      |
  // |----------------+-----------+-----------+---------|
  // | Configuration: | Input     | Input     | Input   |
  // |                | Pull-down | Pull-down | Pull-up |
```

Reply:
Good catch, the comment was correct but the code was wrong, fixed.